### ISSUE TITLE
hxtelemetry data is empty (cause: two threads' StackContext get the same mThreadId of 0)

### ISSUE THREAD

**Issue by jcward** [HaxeFoundation/hxcpp#567]
With Haxe 3.4 and hxcpp 3.4.49, hxtelemetry isn't working as it depends on threadid. (Note that the crash #554 should be fixed by #566, but telemetry data is still empty, hence, this issue.)

Details here: https://github.com/jcward/hxtelemetry/issues/49#issuecomment-280244325

I've hacked around to prove that <strike>ThreadID</strike> **the mThreadId of the StackContexts** is the culprit, but I don't know a reliable fix.

I'll try to add a minimal testcase to this thread for easy reproduction / resolution.

**Comment by jcward** [comment]
Ok, I have a testcase, but it requires this addition to Debug.cpp line 216:

```
void StackContext::onThreadAttach()
{
   #ifdef HXCPP_STACK_IDS
   mThreadId = __hxcpp_GetCurrentThreadNumber();
   printf("Setting stackContext %010x as mThreadId %d\n", this, mThreadId); // <---

   sStackMapMutex.Lock();
   sStackMap[mThreadId] = this;
```

Then run this test:

**build.hxml**
```
-debug
-main Test.hx
-D HXCPP_STACK_TRACE
-cpp out
```

**Test.hx**
```
package;

import cpp.vm.Thread;

class Test
{
  public static function main() { new Test(); }

  public function new()
  {
    var main_thread_id = untyped __global__.__hxcpp_GetCurrentThreadNumber();
    trace("Main thread, my ThreadId = "+main_thread_id);

    trace("Creating _writer thread...");

    var writer = Thread.create(start_writer);

    Sys.sleep(2.0);
    trace("Main thread exit");
  }

  private static function start_writer():Void
  {
    var writer_thread_id = untyped __global__.__hxcpp_GetCurrentThreadNumber();
    trace("Writer thread, my ThreadId = "+writer_thread_id);
  }

}
```

The output is:
```
>./out/Test-debug 
Setting stackContext 00023223f0 as mThreadId 0
Test.hx:12: Main thread, my ThreadId = 0
Test.hx:14: Creating _writer thread...
Setting stackContext 00f00008c0 as mThreadId 0
Test.hx:25: Writer thread, my ThreadId = 1
Test.hx:19: Main thread exit
```

This shows that, while the ThreadID's of the threads are different, at the time the StackContext is instantiated, the mThreadId (used to store the context in sStackMap) is the same for both -- 0. So the stackContext reference stored in the map is overwritten.

So when I try to lookup the stackContext later to [dump telemetry](https://github.com/HaxeFoundation/hxcpp/blob/master/src/hx/Telemetry.cpp#L605-L607), it gets the wrong one.

**Comment by jcward** [comment]
Thank you, @hughsando, this appears to be fixed by 65a2340!

![image](https://cloud.githubusercontent.com/assets/2192439/23152623/5e992b0a-f7c0-11e6-8aaf-a81f1c61739f.png)
